ui/src/workflow/common/NodeContainer.vue

```diff
@@ -152,8 +152,19 @@ const height = ref<{
 })
 const showAnchor = ref<boolean>(false)
 const anchorData = ref<any>()
-const condition = ref('AND')
 
+const condition = computed({
+  set: (v) => {
+    set(props.nodeModel.properties, 'condition', v)
+  },
+  get: () => {
+    if (props.nodeModel.properties.condition) {
+      return props.nodeModel.properties.condition
+    }
+    set(props.nodeModel.properties, 'condition', 'AND')
+    return true
+  }
+})
 const showNode = computed({
   set: (v) => {
     set(props.nodeModel.properties, 'showNode', v)
```

Comment:
The code you provided has some improvements and optimizations that can be made:

### Improvements:
1. **Computed Property Optimization**: The use of `computed` properties for `condition` and `showNode` is a good practice. This allows Vue to efficiently update these properties when data changes, but it's important that they reflect the state correctly without re-setting values unnecessarily.

2. **String Comparison with Single Equal Sign**: Replace single equals signs (`=`) with triple equal signs (`===`) for strict equality comparisons in methods like `set`. Using three equal signs ensures no type coercion occurs, which makes the comparison clearer and less prone to error.

3. **Code Formatting Consistency**: Ensure consistent spacing around operators, braces, and commas to improve readability. This improves maintainability.

4. **Image Source Path Correction**: Ensure the image paths are correct and accessible relative to the directory where this script runs. If images are located outside the assets folder, adjust their paths accordingly.

### Optimize Conditional Rendering:
Instead of using conditional rendering inside the template directly (using `v-if`), consider restructuring the logic so that conditionally rendered elements remain part of the DOM and only visible based on their conditions. This can sometimes be more efficient than creating or destroying them dynamically during component updates.

Here's an updated version of your code incorporating these suggestions):

```vue
<template>
  <!-- Rest of the template remains unchanged -->

<script lang="ts">
import { defineComponent } from 'vue'
// Import other necessary components here

export default defineComponent({
  setup(props) {
    const set = (...args: any[]) => console.log(args) // Placeholder function
    
    const height = ref<{ top?: number; bottom?: number }>({
      // initialization
    })
    
    const showAnchor = ref(false)
    const anchorData = ref({})
    
    const condition = computed(() => {
      return props.nodeModel.properties?.condition || 'AND'
    })
    
    const showNode = computed(() => {
      return props.nodeModel.properties.showNode ?? true
    })
    
    // Other methods and computed properties...
  }
})
</script>

<!-- Reset CSS and styling -->
<style scoped src="./YourStylesheet.css"></style>
```

This revised approach maintains simplicity while improving robustness for future modifications.